eng/scripts/Cadl-Project-Generate.ps1

```diff
@@ -56,6 +56,7 @@ try {
     NpmInstallForProject $npmWorkingDir
     Write-Host("npx cadl compile $mainCadlFile --emit `"`@azure-tools/cadl-csharp`" --output-path `"$resolvedProjectDirectory`"")
     npx cadl compile $mainCadlFile --emit "@azure-tools/cadl-csharp" --output-path "$resolvedProjectDirectory"
+    if ($LASTEXITCODE) { exit $LASTEXITCODE }
```

Comment:
Better to give out an error message.

Reply:
The `npx` command will print error message, so here I just exit.